packages/voice/src/app/agent/chat.tsx

```diff
@@ -183,11 +183,10 @@ export class ChatRequest {
 
       if (!this.done) {
         const currentTurn = isStartConversationRequest ? value.turns.at(-1) : value;
-        const currentMessage = currentTurn.messages
-          .filter((m: any) => m.kind === 'text')
-          .map((m: any) => m.content)
-          .join(' ');
 
+        // Map non-text fragments to empty strings so that they don't appear in the transcript,
+        // but still introduce newlines to make clear that any earlier content is complete.
+        const currentMessage = currentTurn.messages.map((m: any) => (m.kind === 'text' ? m.content : '')).join('\n');
```

Comment:
Nit: I think Justin didn't like newline and preferred spaces (I do prefer newlines though). Not sure how strongly he felt about it. Also, this might create multiple newlines. Maybe try replacing `\n+` with `\n`? (Not quite sure how regex/replace works in JS).

Reply:
Newlines seem kind of artificial to me, you don't see that in typical English text (unless you are suggesting a paragraph break).

Reply:
Since the scenario is:

```
<AssistantMessage>Let me check on that for you.</AssistantMessage>
[...function calls...]
<AssistantMessage>Here's what I found...</AssistantMessage>
```

I think a paragraph break is reasonably appropriate at the end of each `AssistantMessage`?

Reply:
I can get behind that.